docs: 📚 improve contribution guidelines (#231)
synergy-design-system/synergy-design-system: CONTRIBUTING.md

    - [API](#api)
  - [Contribution workflow](#contribution-workflow)
  - [Semantic release process](#semantic-release-process)
    - [Pull requests and commits](#pull-requests-and-commits)
      - [Assignees and reviewers](#assignees-and-reviewers)
  - [Release changes in framework wrapper packages](#release-changes-in-framework-wrapper-packages)
    - [Example Workflow](#example-workflow)
    - [Automatic Checksum Verification and Update](#automatic-checksum-verification-and-update)
  - [Issue tracking](#issue-tracking)

## Overview

Thank you for investing your time to contribute to our project.
We want to make sure to provide our users with a set of feature rich, high quality and easy to use web-components.
To achieve this, we are using some common conventions when contributing features or components to this repository.

## Getting started

- Familiarize yourself with the [Principles of the Synergy Design System](https://synergy-design-system.github.io/?path=/docs/introduction--docs), which serves as a reference for design guidelines, components, and patterns used in this project. Adhering to these principles will help maintain consistency and a cohesive user experience.

> - Have a look at the [storybook demos](https://synergy-design-system.github.io/?path=/docs/components-syn-button--docs) to get a better understanding of the design system and its components.

Reviewer: ```suggestion
- Have a look at the [Storybook demos](https://synergy-design-system.github.io/?path=/docs/components-syn-button--docs) to get a better understanding of the design system and its components.
```

## Coding conventions

We are enforcing common code conventions via `eslint` and `stylelint`. Please make sure your code meets those conventions by linting your files or your PR may not be approved. You may use your editors eslint integration or run `pnpm lint` in the package that you are working on. Linting of all packages is also possible by issuing `pnpm lint` in the root of this repository.

When writing components, use our defined list of design tokens for spacing, fonts or colors. As a rule of thumb, there should be no need to have a static color anywhere in your css code.

Components must be optimized for accessibility. Please use tools like AXE or Dev-Tools and have a look at the [Web Content Accessibility Guidelines](https://www.w3.org/WAI/standards-guidelines/wcag/) for further information.

### Ensure Accessibility

We use semantic HTML elements, ARIA landmarks, roles, states, and properties in components and storybook stories.

> We implement tests and "mouseless" stories to ensure that interactive elements can be navigated and activated using the keyboard (tab-index) and ensure the correct sequence of navigation.

Reviewer: ```suggestion
We implement tests to ensure that interactive elements can be navigated and activated using the keyboard (tab-index) and ensure the correct sequence of navigation.
```

Currently we don't have any mouseless stories.

When creating pull requests use structured PR titles. The title is generated from the GitHub issue title: the issue templates gurantee that the following structure "`prefix`:`icon` `text`" is used.
When merging do a squash and merge. The PR title is used as the single commit message, which keeps the Git history clean. 
The `prefix` is processed when the PR is merged and decides about the release version. Depending on the title, merged PRs can also create a new Synergy version.

Title conventions for our PRs:

`PREFIX: ICON DESCRIPTION`

ICON and PREFIX are one of the following:

| Icon | Used for      | Commit Prefix | Description                 | Release Type |
| :--: | :------------ | :------------ | :-------------------------- | :----------- |
|  ✨  | Feature       | feat          | Add feature                 | Minor        |
|  🤔  | Bugs          | fix           | Fix a bug                   | Patch        |
|  📚  | Documentation | docs          | Add or update documentation | –            |
|  🔧  | Chore         | chore         | Do something else           | –            |

Examples for valid PR titles:

```
fix: Fix accessability in syn-button

> feat: ✨ Added syn-radio

Reviewer: ```suggestion
feat: ✨ Add syn-radio
```

docs: 📚 Add installation instructions

> chore: 🔧 Improved build scripts

Reviewer: ```suggestion
chore: 🔧 Improve build scripts
```

```

> If a change is breaking, be aware that the string "BREAKING CHANGE" must be part of the final commit message when doing the squash and merge. A change is breaking if a dependant project need to adapt (e.g. API has changed).

Reviewer: ```suggestion
If a change is breaking, be aware that the string "BREAKING CHANGE" must be part of the footer (!) of the final commit message when doing the squash and merge. A change is breaking if a dependant project need to adapt (e.g. API has changed).
```

Before merging make sure that all sections are filled out properly and that all DoD checkmarks are checked.

When committing changes use meaningful commit messages. Always imagine the perspective of an outsider: ask yourself, would he/she understand?

#### Assignees and reviewers

> When opening a PR assign yourself to the PR. Once a core team member is available he/she should assign him-/herself as a reviewer. For each PR it must be considered to involve also a designer, if there are design relevant changes. Design feedback is usually given via Chromatic.

Reviewer: ```suggestion
When opening a PR, assign yourself and everyone who should be involved to the PR, at least one maintaining developer and if there are design related tasks on maintaining designer. Design feedback is usually given via Chromatic.
```
